evalml/model_understanding/graphs.py

```diff
@@ -461,19 +461,20 @@ def partial_dependence(pipeline, X, feature, grid_resolution=100):
         raise ValueError("Pipeline to calculate partial dependence for must be fitted")
     if pipeline.model_family == ModelFamily.BASELINE:
         raise ValueError("Partial dependence plots are not supported for Baseline pipelines")
-    if isinstance(pipeline, evalml.pipelines.ClassificationPipeline):
```

Comment:
Previously, we could pass our pipelines directly to scikit-learn's partial dependence method after a bit of finagling (adding attributes expected). This is no longer true because we're returning Woodwork data structures from predict and scikit-learn doesn't know how to handle that. Thus, I needed to wrap our pipeline in scikit_learn_wrapped_estimator (implemented for ensemble), updating the fields now on this wrapped object. IMO kinda cleaner than previous, where we had to delete these ugly attributes we added just for scikit-learn to operate since we don't care for the wrapped obj afterwards :d

Reply:
Yes, thank you for pointing this out! 

I guess the argument against it was, we didn't want to add attributes that scikit-learn had that might not align with what we have, strictly for the sake of calculating partial dependence, for example. Specifically, `_estimator_type` is a bit weird: it makes sense for scikit-learn since they're expecting estimators, but here we're passing in pipelines... so it doesn't make as much sense? (https://scikit-learn.org/stable/modules/generated/sklearn.inspection.partial_dependence.html#sklearn.inspection.partial_dependence)

I think the new implementation helps a bit: it doesn't require us to temporarily set attributes and then delete them. I revisited again and added these attributes to our scikit-learn wrapper instead. This cleans up the partial dependence code as you suggested, but I'd be happy to file an issue though if you think we should still improve upon this / if it still makes sense to add `_estimator_type` 😁